include/fiction/algorithms/simulation/sidb/sidb_simulation_result.hpp

```diff
@@ -59,6 +63,59 @@ struct sidb_simulation_result
      * The key of the map is the name of the parameter, the element is the value of the parameter.
      */
     std::unordered_map<std::string, std::any> additional_simulation_parameters{};
+    /**
+     * This function calculates the ground state charge distributions from the provided simulation results.
+     * The ground state charge distributions are those with energy closest to the minimum energy found in the simulation
+     * results.
+     *
+     * @note When degenerate states exist, there are multiple ground states with the same energy.
+     *
+     * @return A vector of charge distributions with the minimal energy.
+     */
+    [[nodiscard]] std::vector<charge_distribution_surface<Lyt>> groundstates() const noexcept
+    {
+        std::vector<charge_distribution_surface<Lyt>> groundstate_charge_distributions{};
+        std::set<uint64_t>                            charge_indices{};
+
+        // in case, the charge indices are not updated.
+        auto charge_configurations_copy = charge_distributions;
+
+        // Find all unique charge indices. This is done because simulation results can have multiple identical charge
+        // distributions.
+        for (auto& cds : charge_configurations_copy)
+        {
+            cds.charge_distribution_to_index();
+            charge_indices.insert(cds.get_charge_index_and_base().first);
```

Comment:
If we're relying on the charge indices here, wouldn't that break for > 64 dots?

Reply:
That is a really good point and something we should be aware of. And we had this discussion over a year ago that we need to replace the ``uint64_t`` charge index with something else in the future. 
So far I have been doing simulations for smaller layouts. So we could live with it. However, as @wlambooy takes big steps towards larger layouts, this may become an issue at some point. Typically our SiDB circuit has about 50% negatively charged SiDBs, which means it could break at about 120 dots, right?